docs/enhancements/platform-integration.md

```diff
@@ -0,0 +1,134 @@
+- Allow providers list all enabled UI features via `SetPlatformUsages` Provider interface
+- `platform_type` feature should be validated by assisted-service against version support matrix (i.e. Nutanix platform available for 4.11+ clusters)
+- Allow providers list additional validations to be performed via `SetPlatformUsages` Provider interface
+- Document requirements for openshift-installer
+
+### User Stories
+
+#### Story 1
+
+As an Openshift developer working on provider X I want to add support for installing Openshift with assisted installer
+on top of platform X.
+
+#### Story 2
+
+As an Openshift developer working on the assisted installer team I want to let providers hook into the
+assisted installer flow natural without having to worry about provider specific implementation.
+
+#### Story 3
+
+As an Openshift developer working on the assisted installer UI team I want to use a unified API to enable platform integration feature and use the API to selectively enable or disable features.
+
+### Implementation Details/Notes/Constraints
+
+#### Changes to the assisted-service project:
+
+##### Step 1: Design API for provider-specific features
+
+Extend Usages API to include:
+* additional validations
+* UI features
+* provider's `SetPlatformUsages` should list all enabled UI features.
+  If provider doesn't list a feature its condsidered to be disabled.
+
+#### Changes to the assisted-ui project:
+
+##### Step 1: Rework UI using API from previous step
+
+UI needs to fetch available features enabled via `usages` API and enable/disable elements accordingly
+
+#### Changes to the assisted-agent project:
+
+##### Step 1: Detect provider-specific settings
+
+Providers will list features and additional validations in ``SetPlatformUsages`
+
+#### Changes to the Openshift installer project:
+
+##### Step 1: Document requirements for `create manifest` step
+
+In order to avoid storing sensitive information and credentials, providers are configured using stub
+data in install-config.yaml. User is expected to update this information after installation, via CLI, console or assisted installer UI addon.
+
+`openshift-install create manifests` runs on assisted-service host, so any requests to provider API will fail.
+This step should not require provider API connection to pass, but `create cluster` should be able to.
+
+All validations requiring API access in installer should be moved out of:
+* `pkg/asset/installconfig/<provider name>/validation.go`'s `Validate` to `ValidateForProvisioning` function
+* `pkg/asset/manifests/infrastructure.go`
+
+### Risks and Mitigations
+
+Implementing provider-specific flows would extend our testing matrix and may confuse users. The mitigation may be additional CI tests and feature-gating UI changes.
+
+## Design Details
+
+### Closed Questions
+
+### UI Impact
+
+UI needs to use `usages` API to selectively enable or disable UI features.
```

Comment:
'usages'? I think it should be 'feature supported API

Reply:
Good catch, updated